examples/omnivore/data/datasets.py

```diff
@@ -0,0 +1,221 @@
+            "lecture_theatre",
+            "library",
+            "living_room",
+            "office",
+            "rest_space",
+            "study_space",
+        ]
+        self.class_to_idx = dict(zip(self.classes, range(len(self.classes))))
+
+        allsplit_filepath = self._meta_dir / "traintestSUNRGBD/allsplit.mat"
+        allsplit_mat = scipy.io.loadmat(allsplit_filepath)
+
+        # The original filepath on the "allsplit.mat" has the prefix from author machine that need to be replaced
+        ori_prefix = "/n/fs/sun3d/data/SUNRGBD/"
+        if split == "train":
+            self.image_dirs = [
+                self._data_dir / x[0][len(ori_prefix) :]
+                for x in allsplit_mat["alltrain"][0]
+            ]
+        elif split == "val":
+            self.image_dirs = [
+                self._data_dir / x[0][len(ori_prefix) :]
+                for x in allsplit_mat["alltest"][0]
+            ]
+
+        # Filter to use only chosen 19 classes
+        self.image_dirs = [
+            x
+            for x in self.image_dirs
+            if self._get_sunrgbd_scene_class(x) in self.class_to_idx
+        ]
+
+    def _check_exists(self):
+        return self._data_dir.is_dir() and self._meta_dir.is_dir()
+
+    def __len__(self):
+        return len(self.image_dirs)
+
+    def _get_disparity_tensor(self, image_dir):
+        # Using depth_bfx, but maybe can also consider just using depth
+        image_dir = Path(image_dir)
+        depth_dir = image_dir / "depth_bfx"
+        intrinsics_file = image_dir / "intrinsics.txt"
+        depth_path = depth_dir / os.listdir(depth_dir)[0]
+
+        sensor_type = image_dir.relative_to(self._data_dir).parts[0]
+        baseline = self.sensor_to_params[sensor_type]["baseline"]
+        with open(intrinsics_file, "r") as fin:
+            lines = fin.readlines()
+            focal_length = float(lines[0].strip().split()[0])
+
+        img_depth = PIL.Image.open(depth_path)
+        tensor_depth = T.ToTensor()(img_depth)
+        tensor_disparity = baseline * focal_length / (tensor_depth / 1000.0)
+        return tensor_disparity
+
+    def _read_sunrgbd_image(self, image_dir):
+        rgb_dir = os.path.join(image_dir, "image")
+        rgb_path = os.path.join(rgb_dir, os.listdir(rgb_dir)[0])
+        img_rgb = PIL.Image.open(rgb_path)
+        tensor_rgb = T.ToTensor()(img_rgb)
+
+        tensor_d = self._get_disparity_tensor(image_dir)
+
+        tensor_rgbd = torch.cat((tensor_rgb, tensor_d), dim=0)
+        return tensor_rgbd
+
+    def _get_sunrgbd_scene_class(self, image_dir):
+        with open(os.path.join(image_dir, "scene.txt"), "r") as fin:
+            scene_class = fin.read().strip()
+        return scene_class
+
+    def __getitem__(self, idx):
+        # return tuple of image (H W C==4) and scene class index
+        image_dir = self.image_dirs[idx]
+        x_rgbd = self._read_sunrgbd_image(image_dir)
+        scene_class = self._get_sunrgbd_scene_class(image_dir)
```

Comment:
I think it will not be useful to cache since `image_dir` is directory of a single image that contain the rgb image, depth_image, and scene name in scene.txt.
Example of scene.txt contain of text like `bedroom`.
Since each `idx` will have different `image_dir`, therefore caching doesn't seems efficient in this case.